core/src/main/java/com/yahoo/imapnio/async/data/QResyncSeqMatchData.java

```diff
@@ -0,0 +1,61 @@
+package com.yahoo.imapnio.async.data;
+
+import java.util.List;
+
+import javax.annotation.Nonnull;
+
+/**
+ * This class models the QRESYNC Sequence match data.
+ */
+public class QResyncSeqMatchData {
+    /** Message sequence number. */
+    private List<MessageNumberSet> messageSeqNumbers;
+
+    /** Corresponding UIDs. */
+    private List<MessageNumberSet> uids;
+
+    /**
+     * Constructor.
+     * @param messageNumberSet message sequence numbers
+     * @param uid UIDs
+     */
+    public QResyncSeqMatchData(@Nonnull final List<MessageNumberSet> messageNumberSet, @Nonnull final List<MessageNumberSet> uid) {
+        this.messageSeqNumbers = messageNumberSet;
+        this.uids = uid;
+    }
+
+    /**
+     * Get the message sequence number.
+     * @return message sequence numbers
+     */
+    public List<MessageNumberSet> getMessageSeqNumbers() {
+        return messageSeqNumbers;
+    }
+
+    /**
+     * Get the uid numbers.
+     * @return UIDs
+     */
+    public List<MessageNumberSet> getUids() {
+        return uids;
+    }
+
+    /**
+     * Convert to command line string.
+     * @return String
+     */
+    @Override
+    public String toString() {
+        StringBuilder sb = new StringBuilder();
+        if (messageSeqNumbers != null && messageSeqNumbers.size() > 0) {
```

Comment:
Why messageSeqNumbers != null check? This params is set to not null

Reply:
setting parameter annotation does not make sure it will be not null.
Just additional check to make sure it doesn't throw exception.

Reply:
True. But that's the contract this class sets. So the check defeats the purpose.